src/test/java/org/opensearch/security/dlic/rest/api/DlsRoleSearchFilteringTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ */
+
+package org.opensearch.security.dlic.rest.api;
+
+import java.util.stream.IntStream;
+
+import org.apache.hc.core5.http.HttpStatus;
+import org.junit.Test;
+
+import org.opensearch.security.test.AbstractSecurityUnitTest;
+import org.opensearch.security.test.helper.rest.RestHelper;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.opensearch.security.OpenSearchSecurityPlugin.PLUGINS_PREFIX;
+
+public class DlsRoleSearchFilteringTest extends AbstractRestApiUnitTest {
+    private final String SECURITY_ENDPOINT;
+
+    public DlsRoleSearchFilteringTest() {
+        SECURITY_ENDPOINT = PLUGINS_PREFIX + "/api";
+    }
+
+    @Test
+    public void testRoleUnionSearchFiltering() throws Exception {
+        setup();
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        // create roles
+        final String ROLE_ENDPOINT = SECURITY_ENDPOINT + "/roles";
+        RestHelper.HttpResponse testRoleOneResponse = rh.executePutRequest(
+            ROLE_ENDPOINT + "/test_role_1",
```

Comment:
Can these roles be shortened to only the permissions needed for the test. i.e. only index permissions on the indices needed. It looks like the `tenant_permissions` portions of these roles are not necessary for the test.

Its hard to read the permissions in the test since the JSON is escaped and on a single line.